### ISSUE TITLE
Runtime panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by xuyenvuong** [prysmaticlabs/prysm#13086]
### Describe the bug

Running Mainnet with
`--enable-experimental-state`
`--enable-eip-4881`

then Beacon was stopped responding.  Restart beacon/geth couple times and the issue was still persisted. 

### Has this worked before in a previous version?

```text
Not seeing this issue with previous version.
```

### 🔬 Minimal Reproduction

Run beacon with these flags for awhile, beacon stop responding
`--enable-experimental-state`
`--enable-eip-4881`

Then proceed to remove the above two flags, the error start to show right after the beacon restart. Then beacon reload itself.

### Error

```text
Oct 20 20:41:21 node prysm.sh[6858]: time="2023-10-20 20:41:21" level=info msg="Running node with peer id of 16Uiu2HAmNnzfe1fpUeanaXTZVrULqtZYYtTNxeF5NLpV5Tt3uFkW " prefix=p2p
Oct 20 20:41:22 node prysm.sh[6858]: time="2023-10-20 20:41:22" level=error msg="Runtime panic: runtime error: invalid memory address or nil pointer dereference
Oct 20 20:41:22 node prysm.sh[6858]: goroutine 1 [running]:
Oct 20 20:41:22 node prysm.sh[6858]: runtime/debug.Stack()
Oct 20 20:41:22 node prysm.sh[6858]:         GOROOT/src/runtime/debug/stack.go:24 +0x65
Oct 20 20:41:22 node prysm.sh[6858]: main.main.func3()
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:215 +0x45
Oct 20 20:41:22 node prysm.sh[6858]: panic({0x440840, 0x35c5590})
Oct 20 20:41:22 node prysm.sh[6858]:         GOROOT/src/runtime/panic.go:884 +0x213
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/container/trie.CreateTrieFromProto(0x0)
Oct 20 20:41:22 node prysm.sh[6858]:         container/trie/sparse_merkle.go:34 +0x33
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/execution.(*Service).initializeEth1Data(0xc0008ba380, {0xdd3cd0, 0xc0267c3b80}, 0xc0004e39d0)
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/execution/service.go:761 +0x10d
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/execution.NewService({0xdd3cd0?, 0xc0008d4000?}, {0xc0306717a0, 0xb, 0x418aa0?})
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/execution/service.go:219 +0x6c7
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/node.(*BeaconNode).registerPOWChainService(0xc000158a80)
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/node/node.go:674 +0x4a9
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/node.New(0xc0002e3340, {0xc0003f9110, 0x5, 0x0?})
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/node/node.go:233 +0x1045
Oct 20 20:41:22 node prysm.sh[6858]: main.startNode(0x1d35767?)
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:292 +0x591
Oct 20 20:41:22 node prysm.sh[6858]: main.main.func1(0xc000a14160?)
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:147 +0x19
Oct 20 20:41:22 node prysm.sh[6858]: github.com/urfave/cli/v2.(*Command).Run(0xc000a14160, 0xc0002e3340, {0xc0000520c0, 0x4, 0x4})
Oct 20 20:41:22 node prysm.sh[6858]:         external/com_github_urfave_cli_v2/command.go:274 +0x9eb
Oct 20 20:41:22 node prysm.sh[6858]: github.com/urfave/cli/v2.(*App).RunContext(0xc000a0c000, {0xdd3d08?, 0xc000054058}, {0xc0000520c0, 0x4, 0x4})
Oct 20 20:41:22 node prysm.sh[6858]:         external/com_github_urfave_cli_v2/app.go:332 +0x616
Oct 20 20:41:22 node prysm.sh[6858]: github.com/urfave/cli/v2.(*App).Run(...)
Oct 20 20:41:22 node prysm.sh[6858]:         external/com_github_urfave_cli_v2/app.go:309
Oct 20 20:41:22 node prysm.sh[6858]: main.main()
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:220 +0x1f3
Oct 20 20:41:22 node prysm.sh[6858]: " prefix=main
Oct 20 20:41:22 node prysm.sh[6858]: panic: runtime error: invalid memory address or nil pointer dereference [recovered]
Oct 20 20:41:22 node prysm.sh[6858]:         panic: runtime error: invalid memory address or nil pointer dereference
Oct 20 20:41:22 node prysm.sh[6858]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x2565e93]
Oct 20 20:41:22 node prysm.sh[6858]: goroutine 1 [running]:
Oct 20 20:41:22 node prysm.sh[6858]: main.main.func3()
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:216 +0xd4
Oct 20 20:41:22 node prysm.sh[6858]: panic({0x440840, 0x35c5590})
Oct 20 20:41:22 node prysm.sh[6858]:         GOROOT/src/runtime/panic.go:884 +0x213
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/container/trie.CreateTrieFromProto(0x0)
Oct 20 20:41:22 node prysm.sh[6858]:         container/trie/sparse_merkle.go:34 +0x33
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/execution.(*Service).initializeEth1Data(0xc0008ba380, {0xdd3cd0, 0xc0267c3b80}, 0xc0004e39d0)
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/execution/service.go:761 +0x10d
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/execution.NewService({0xdd3cd0?, 0xc0008d4000?}, {0xc0306717a0, 0xb, 0x418aa0?})
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/execution/service.go:219 +0x6c7
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/node.(*BeaconNode).registerPOWChainService(0xc000158a80)
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/node/node.go:674 +0x4a9
Oct 20 20:41:22 node prysm.sh[6858]: github.com/prysmaticlabs/prysm/v4/beacon-chain/node.New(0xc0002e3340, {0xc0003f9110, 0x5, 0x0?})
Oct 20 20:41:22 node prysm.sh[6858]:         beacon-chain/node/node.go:233 +0x1045
Oct 20 20:41:22 node prysm.sh[6858]: main.startNode(0x1d35767?)
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:292 +0x591
Oct 20 20:41:22 node prysm.sh[6858]: main.main.func1(0xc000a14160?)
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:147 +0x19
Oct 20 20:41:22 node prysm.sh[6858]: github.com/urfave/cli/v2.(*Command).Run(0xc000a14160, 0xc0002e3340, {0xc0000520c0, 0x4, 0x4})
Oct 20 20:41:22 node prysm.sh[6858]:         external/com_github_urfave_cli_v2/command.go:274 +0x9eb
Oct 20 20:41:22 node prysm.sh[6858]: github.com/urfave/cli/v2.(*App).RunContext(0xc000a0c000, {0xdd3d08?, 0xc000054058}, {0xc0000520c0, 0x4, 0x4})
Oct 20 20:41:22 node prysm.sh[6858]:         external/com_github_urfave_cli_v2/app.go:332 +0x616
Oct 20 20:41:22 node prysm.sh[6858]: github.com/urfave/cli/v2.(*App).Run(...)
Oct 20 20:41:22 node prysm.sh[6858]:         external/com_github_urfave_cli_v2/app.go:309
Oct 20 20:41:22 node prysm.sh[6858]: main.main()
Oct 20 20:41:22 node prysm.sh[6858]:         cmd/beacon-chain/main.go:220 +0x1f3
```

### Platform(s)

Linux (x86)

### What version of Prysm are you running? (Which release)

v4.1.0

**Comment by xuyenvuong** [comment]
I tried to put these two flags back, and the same errors were not shown anymore.
--enable-experimental-state
--enable-eip-4881: true

**Comment by xuyenvuong** [comment]
OK once this is happening, the only solution for me was remove the whole beaconchain db, and disable these 2 flags:
`--enable-experimental-state`
`--enable-eip-4881` 
then everything start to work again.

**Comment by prestonvanloon** [comment]
Seems like the panic was resolved by #13096
